met-api/migrations/versions/a1237c8a3df9_.py

```diff
@@ -0,0 +1,46 @@
+"""
+
+Revision ID: a1237c8a3df9
+Revises: 3f11e9145a08
+Create Date: 2023-03-07 12:19:32.133932
+
+"""
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql
+from datetime import datetime
+
+# revision identifiers, used by Alembic.
+revision = 'a1237c8a3df9'
+down_revision = '3f11e9145a08'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    widget_map = op.create_table('widget_map',
+    sa.Column('id', sa.Integer(), autoincrement=True, nullable=False),
+    sa.Column('description', sa.String(length=500), nullable=True),
+    sa.Column('longitude', sa.Float(), nullable=False),
+    sa.Column('latitude', sa.Float(), nullable=False),
+    sa.Column('engagement_id', sa.Integer(), nullable=True),
+    sa.Column('widget_id', sa.Integer(), nullable=True),
+    sa.Column('created_date', sa.DateTime(), nullable=False),
+    sa.Column('updated_date', sa.DateTime(), nullable=True),
+    sa.Column('created_by', sa.String(length=50), nullable=True),
+    sa.Column('updated_by', sa.String(length=50), nullable=True),
+    sa.ForeignKeyConstraint(['widget_id'], ['widget.id'], ondelete='CASCADE'),
```

Comment:
Not sure i unnderstand this foreign key structure, If I remember correctly our Widget and Widget item tables contains a widget_data_id and we usually use that column to provide. the item id (in this case the widgetmap.id), did something change?

Reply:
Yeah I think we haven't followed that same pattern with documents and events. Widget_item could be refactored to like widget_contact. Instead of using widget_item with its data id, we created widget_events or widget_documents and just had the widge_id there.